sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -683,47 +697,47 @@ void Bucket::freeSlab(Slab &Slab, bool &ToPool) {
   std::lock_guard<std::mutex> Lg(BucketLock);
   auto SlabIter = Slab.getIterator();
   assert(SlabIter != UnavailableSlabs.end());
-  if (CanPool()) {
+  if (CanPool(ToPool)) {
     auto It =
         AvailableSlabs.insert(AvailableSlabs.begin(), std::move(*SlabIter));
     UnavailableSlabs.erase(SlabIter);
     (*It)->setIterator(It);
-
-    if (USMSettings.PoolTrace > 1) {
-      updateStats(-1, 1);
-      ToPool = true;
-    }
   } else {
     UnavailableSlabs.erase(SlabIter);
-
-    if (USMSettings.PoolTrace > 1) {
-      updateStats(-1, 0);
-      ToPool = false;
-    }
   }
 }
 
-auto Bucket::getAvailSlab(bool &FromAllocatedSlab)
+auto Bucket::getAvailSlab(bool &FromSlab, bool &FromPool)
```

Comment:
Can FromSlab and FromPool be members of the Slab class?

Reply:
So, I suppose there is some reason not to do this? Can you please elaborate a bit? Thanks.

Reply:
The output values FromSlab and FromPool are used at the outermost level to note how a particular allocation request was satisfied - whether it was from a pooled slab or from a fresh USM allocation, and for chunked allocations, whether from an empty slab or a partially filled slab. At the outermost level of the function call hierarchy we don't know anything about Buckets and Slabs, so keeping these flags in a Slab struct would not be helpful. Secondly, these values change with each allocation request. They are not tied to a slab. From the same slab, depending on the memory allocation pattern, we could return different values of these two flags.